test/decimal.test.js

```diff
@@ -63,4 +69,431 @@ describe('model with decimal property', function() {
       })
       .catch(done);
   });
+
+  context('nested decimal props', function() {
+    it('should create/update instance for array of decimal props', function(done) {
+      modelWithDecimalArray = db.define('modelWithDecimalArray', {
+        randomReview: {
+          type: [String],
+          mongodb: {
+            dataType: 'Decimal128',
+          },
+        },
+      }, {
+        updateOnLoad: true,
+      });
+
+      var createData = {
+        'randomReview': [
+          '3.5',
+          '4.5',
+          '4.0',
+        ],
+      };
+      var updateData = {
+        'randomReview': [
+          '5.5',
+          '5.5',
+          '5.5',
+        ],
+      };
+      let instanceId;
+
+      modelWithDecimalArray.create(createData)
+        .then(function(inst) {
+          instanceId = inst.id;
+          return findModelInstanceAsync('modelWithDecimalArray', instanceId);
+        })
+        .then(function(createdInstance) {
+          createdInstance.randomReview[0].should.be.instanceOf(Decimal128);
+          createdInstance.randomReview[0].should.deepEqual(Decimal128.fromString('3.5'));
+          return modelWithDecimalArray.updateAll({id: instanceId}, updateData);
+        })
+        .then(function(inst) {
+          return findModelInstanceAsync('modelWithDecimalArray', instanceId);
+        })
+        .then(function(updatedInstance) {
+          updatedInstance.randomReview[0].should.be.instanceOf(Decimal128);
+          updatedInstance.randomReview[0].should.deepEqual(Decimal128.fromString('5.5'));
+          done();
```

Comment:
Don't mix callbacks with promises. Mocha supports tests written with promises, use that feature please.

```ts
it('should create/update instance for array of decimal props', function() {
  // ...
  return modelWithDecimalArray.create(createData)
    // ...
   .then(function(updatedInstance) {
      updatedInstance.randomReview[0].should.be.instanceOf(Decimal128);
      updatedInstance.randomReview[0].should.deepEqual(Decimal128.fromString('5.5'));
    });
});
```

The same comment applies to other tests too.